vm/systemSmartContracts/delegation.go

```diff
@@ -568,21 +590,27 @@ func (d *delegation) removeNodes(args *vmcommon.ContractCallInput) vmcommon.Retu
 }
 
 func (d *delegation) stakeNodes(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	returnCode := d.checkOwnerCallValueGas(args)
+	returnCode := d.checkOwnerCallValueGasAndDuplicates(args)
 	if returnCode != vmcommon.Ok {
 		return returnCode
 	}
 	if len(args.Arguments) == 0 {
 		d.eei.AddReturnMessage("not enough arguments")
 		return vmcommon.FunctionWrongSignature
 	}
+	//TODO: Shouldn't be checked if there is enough gas for all the nodes which would be staked?
+	err := d.eei.UseGas(uint64(len(args.Arguments)) * d.gasCost.MetaChainSystemSCsCost.DelegationOps)
```

Comment:
this check is done inside useGas -> it returns not enough gas. No need for further check

Reply:
But this call is missing at all in original branch